frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -406,37 +405,74 @@ export const removeResourceConnection = (
   });
 };
 
+const safeKill = async (model: K8sKind, obj: K8sResourceKind) => {
+  const resp = await checkAccess({
+    group: model.apiGroup,
+    resource: model.plural,
+    verb: 'delete',
+    name: obj.metadata.name,
+    namespace: obj.metadata.namespace,
+  });
+  if (resp.status.allowed) {
+    return k8sKill(model, obj);
+  }
+  return null;
+};
+
+const deleteWebhooks = (
+  resource: K8sResourceKind,
+  workload: TopologyDataObject<{ isKnativeResource?: boolean }>,
+) => {
+  const isKnativeResource = workload?.data?.isKnativeResource ?? false;
+  const deploymentsAnnotations = resource.metadata?.annotations ?? {};
+  const gitType = detectGitType(deploymentsAnnotations['app.openshift.io/vcs-uri']);
+  const buildConfigs = workload?.resources?.buildConfigs;
+  return buildConfigs?.reduce((requests, bc) => {
+    const triggers = bc.spec?.triggers;
+    const reqs = triggers?.reduce((a, t) => {
+      let obj: K8sResourceKind;
+      if (t.generic) {
+        obj = {
+          ...resource,
+          metadata: {
+            name:
+              t.generic?.secretReference?.name ??
+              `${resource.metadata.name}-generic-webhook-secret`,
+            namespace: resource.metadata.namespace,
+          },
+        };
+      }
+      if (t[gitType] && !isKnativeResource) {
+        obj = {
+          ...resource,
+          metadata: {
+            name:
+              t[gitType]?.secretReference?.name ??
+              `${resource.metadata.name}-${gitType}-webhook-secret`,
+            namespace: resource.metadata.namespace,
+          },
+        };
+      }
```

Comment:
```suggestion
     const webhookType = t.generic ? 'generic' : gitType; 
      const webhookTypeObj = t.generic || (!isKnativeResource && t[gitType])
      if (webhookTypeObj) {
        obj = {
          ...resource,
          metadata: {
            name:
              webhookTypeObj.secretReference?.name ??
              `${resource.metadata.name}-${webhookType}-webhook-secret`,
            namespace: resource.metadata.namespace,
          },
        };
      }
```
@sahil143 something like this can be done to reduce the lines of code here because both the objects are similar except for the name. WDYT?

Reply:
yeah, for these two cases make sense. updated